components/script/dom/bindings/codegen/CodegenRust.py

```diff
@@ -3642,23 +3638,33 @@ def defineJitInfo(self, infoName, opName, opType, infallible, movable,
         def jitInfoInitializer(isTypedMethod):
             initializer = fill(
                 """
-                JSJitInfo {
-                    call: ${opName} as *const os::raw::c_void,
-                    protoID: PrototypeList::ID::${name} as u16,
-                    depth: ${depth},
-                    _bitfield_1:
-                        JSJitInfo::new_bitfield_1(
-                            JSJitInfo_OpType::${opType} as u8,
-                            JSJitInfo_AliasSet::${aliasSet} as u8,
-                            JSValueType::${returnType} as u8,
-                            ${isInfallible},
-                            ${isMovable},
-                            ${isEliminatable},
-                            ${isAlwaysInSlot},
-                            ${isLazilyCachedInSlot},
-                            ${isTypedMethod},
-                            ${slotIndex} as u16,
-                        )
+                jsapi::JSJitInfo {
+                    __bindgen_anon_1: jsapi::JSJitInfo__bindgen_ty_1 {
+                        method: unsafe {
+                            mem::transmute(${opName} as
+                                unsafe extern "C" fn(*mut jsapi::JSContext,
+                                                     jsapi::JS::HandleObject,
+                                                     _,
+                                                     _) -> bool) },
```

Comment:
Why is this transmute required?

Reply:
It's there to change the `*const ActualType` parameter into a `*const std::os::raw::c_void`. Upon reflection this is better done with `as`.

Reply:
hmmm nevermind, `as` doesn't work here:

```
error[E0605]: non-primitive cast: `unsafe extern "C" fn(*mut js::jsapi::JSContext, js::jsapi::JS::Handle<*mut js::jsapi::JSObject>, *const dom::xmlhttprequesteventtarget::XMLHttpRequestEventTarget, js::jsapi::JSJitSetterCallArgs) -> bool {dom::bindings::codegen::Bindings::XMLHttpRequestEventTargetBinding::XMLHttpRequestEventTargetBinding::set_onloadend}` as `unsafe extern "C" fn(*mut js::jsapi::JSContext, js::jsapi::JS::Handle<*mut js::jsapi::JSObject>, *mut std::os::raw::c_void, *const js::jsapi::JSJitMethodCallArgs) -> bool`
   --> /home/fitzgen/servo/target/debug/build/script-11a815433a47bb39/out/Bindings/XMLHttpRequestEventTargetBinding.rs:885:22
    |
885 |           method: Some(set_onloadend as
    |  ______________________^
886 | |             unsafe extern "C" fn(*mut jsapi::JSContext,
887 | |                                  jsapi::JS::Handle<*mut jsapi::JSObject>,
888 | |                                  _,
889 | |                                  _) -> bool),
    | |___________________________________________^
    |
    = note: an `as` expression can only be used to convert between primitive types. Consider using the `From` trait
```